### PR TITLE
fix: Eval of params with missing src dep and no default value

### PR DESCRIPTION
**Pull request by daniel-codefresh**
Signed-off-by: Daniel Soifer <daniel.soifer@codefresh.io>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by daniel-codefresh** [comment]
Fixes: #1366

File: sensors/triggers/params.go [hunk 1]
```diff
@@ -159,60 +162,71 @@ func renderEventDataAsJSON(event *v1alpha1.Event) ([]byte, error) {
...
-			return *src.Value, nil
+			return src.Value, nil
 		}
-		return string(value), nil
+
+		strPayload := string(eventPayload)
+		return &strPayload, nil
 	}
-	return "", errors.Wrapf(err, "unable to resolve '%s' parameter value", src.DependencyName)
+	return nil, errors.Wrapf(err, "unable to resolve '%s' parameter value", src.DependencyName)
```
Comment (whynowy) [inline sensors/triggers/params.go:229]: It seems `err` is always `nil`, shall this be changed to:

```
return nil, fmt.Errorf("unable to resolve '%s' parameter value", src.DependencyName)
```

Comment (daniel-codefresh) [reply sensors/triggers/params.go:229]: Yes 100%, just forgot to change this line

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**